src/bindgen/ir/structure.rs

```diff
@@ -24,6 +24,7 @@ pub struct Struct {
     pub generic_params: Vec<String>,
     pub documentation: Documentation,
     pub functions: Vec<Function>,
+    pub destructor: Option<Function>,
```

Comment:
I'm worried this could be a potential footgun. In most cases if you implement any of the copy constructor, copy assignment, or destructor, you need to implement them all. [See the rule of three](http://en.cppreference.com/w/cpp/language/rule_of_three).

For example
```rust
struct WrapperVec {
  data: Vec<u8>,
}
fn destructor(this: *WrapperVec) {
  mem::drop(*this)
}
```
```c++
{
  WrapperVec a = ...
  WrapperVec b = a; // default copy constuctor does a shallow copy of the vec ptr's
} // destructor for a and b runs and the vec is dropped twice leading to a double free
```
This is a tricky problem. If we were to disable the copy constructor and copy assignment and only allow moves, I think we'd be in a better position. That would disadvantage rust structs that truly are POD. We could maybe generate `clone()` methods then?

I think whatever the solution is, it needs to be complete so we can't just generate destructors.